### ISSUE TITLE
System.OutOfMemoryException

### ISSUE THREAD

**Issue by pelstob** [Karlovsky120/7DaysProfileEditor#28]
After replacing the executable of the last version it wasn't running at all.  If I delete 7DaysProfileEditor.config and re-launch I can get it to work once, but it pops up with an error stating "Error while processing icons. Failed to load asset files.Exception of type 'System.OutOfMemoryException' was thrown."  Any subsequent runs and nothing appears to happen other than an entry in eventvwr:

```
Application: 7DaysProfileEditor.exe
Framework Version: v4.0.30319
Description: The process was terminated due to an unhandled exception.
Exception Info: System.AccessViolationException
   at System.Runtime.InteropServices.Marshal.CopyToManaged(IntPtr, System.Object, Int32, Int32)
   at SevenDaysProfileEditor.Data.IconData.GetTextureAtlas(System.IO.BinaryReader, Int32, Int32)
   at SevenDaysProfileEditor.Data.IconData.ReadDefaultItemIcons()
   at SevenDaysProfileEditor.Program.Initialize()
   at SevenDaysProfileEditor.Program.Start(SevenDaysProfileEditor.GUI.MainWindow)
   at SevenDaysProfileEditor.Program.Main()
```

ProfileEditor.log contents.

```
-----------------------------------------------------
Log started at: 02/09/2017 23:09:40
-----------------------------------------------------

Time Stamp: 02/09/2017 23:09:41
Message: Exception of type 'System.OutOfMemoryException' was thrown.
Source: 7DaysProfileEditor
Exception: System.OutOfMemoryException: Exception of type 'System.OutOfMemoryException' was thrown.
   at SevenDaysProfileEditor.Util.ReadAssetString(BinaryReader reader, Int32 length)
   at SevenDaysProfileEditor.Data.IconData.ReadUIIcons()
   at SevenDaysProfileEditor.Program.Initialize()
Target Site: System.String ReadAssetString(System.IO.BinaryReader, Int32)





Message: Error while processing icons. Failed to load asset files.Exception of type 'System.OutOfMemoryException' was thrown.
-------------------------------------------------------------------
```

**Comment by madmunki** [comment]
Out of memory, whats your system memory? Can you try rebooting to free up memory?

**Comment by pelstob** [comment]
I'm reasonable sure it's not me; I've rebooted and at the time of running I had >10GiB free RAM of 16GiB total.

The out of memory error will possibly be a byproduct of Windows protecting the system from the access violation.

**Comment by madmunki** [comment]
I’ll check out the code in this area.

**Comment by MrSkyFire** [comment]
This program worked the first time, then stopped working with this error.
Full Error: 
"Error while processing icons. Failed to load asset files. Exception of type 'System.OutOfMemoryException' was thrown."

**Comment by DorHans** [comment]
Well, I get a "System.IO.EndOfStreamException", but it could be the same reason.
I traced this error to the file "7DaysProfileEditor/7DaysProfileEditor/source/Data/IconData.cs". The method "private static void ReadUIIcons()" tries to read the icons for the skills. In this method you access an asset file by its index: `AssetInfo uiIconText = AssetInfo.GetAssetInfoByIndex(43116);`
But for me, this file has an index of 59162. This asset is a "MonoBehaviour" (0xFFFFFFF1), but it gives me perfect icons, and no exceptions. :)

I don't know why the index changed, but maybe that's the problem. If your file has a different index, you may end up reading so much stuff, it completely eats your memory.

**Comment by Karlovsky120** [comment]
Something has changed indeed, but the plan for a future update is to just bundle the default icons with the program, so hat this or similar scenarios do not occur.

**Comment by madmunki** [comment]
Need a way to reliably get the offset. If we change it from 43116 to 59162, then it breaks reading older files. 

Thanks for finding the new offset, ill update it until we have a better solution.

**Comment by madmunki** [comment]
Resolved in 0.16.3.3.
